tests/feature_selection/test_relevance.py

```diff
@@ -0,0 +1,133 @@
+# -*- coding: utf-8 -*-
+# This file as well as the whole tsfresh package are licenced under the MIT licence (see the LICENCE.txt)
+# Maximilian Christ (maximilianchrist.com), Blue Yonder Gmbh, 2016
+
+import numpy as np
+import pandas as pd
+import pytest
+import mock
+
+from tsfresh.feature_selection.relevance import infer_ml_task, calculate_relevance_table, combine_relevance_tables, \
+    get_feature_type
+
+
+class TestInferMLTask:
+    def test_infers_classification_for_integer_target(self):
+        y = pd.Series([1, 2, 3])
+        assert 'classification' == infer_ml_task(y)
+
+
+    def test_infers_classification_for_boolean_target(self):
+        y = pd.Series([True, False, False])
+        assert 'classification' == infer_ml_task(y)
+
+    def test_infers_classification_for_object_target(self):
+        y = pd.Series(['high', 'low'])
+        assert 'classification' == infer_ml_task(y)
+
+    def test_infers_regression_for_float_target(self):
+        y = pd.Series([1.0, 1.5, 1.7])
+        assert 'regression' == infer_ml_task(y)
+
+
+class TestCalculateRelevanceTable:
+    @pytest.fixture()
+    def y_binary(self):
+        return pd.Series([0, 1, 1])
+
+    @pytest.fixture()
+    def y_real(self):
+        return pd.Series([0.1, 0.2, 0.1])
+
+    @pytest.fixture()
+    def X(self):
+        df = pd.DataFrame()
+        df['feature_binary'] = [1, 1, 0]
+        df['feature_real'] = [0.1, 0.2, 0.3]
+        return df
+
+    def test_restrict_ml_task_options(self, X, y_binary):
+        with pytest.raises(ValueError):
+            calculate_relevance_table(X, y_binary, ml_task='some_other_task')
+
+    def test_constant_feature_irrelevant(self, y_binary):
+        X = pd.DataFrame([1, 1, 1], columns=['feature_binary'])
+
+        relevance_table = calculate_relevance_table(X, y_binary)
+        assert "feature_binary" == relevance_table.index[0]
```

Comment:
I'm not sure there is an actual difference. `pytest` rewrites `assert` statements before executing them. See https://docs.pytest.org/en/latest/assert.html#assert for more details.
For me using the plain assert makes the tests much clearer.